store/src/commonMain/kotlin/com/dropbox/external/store5/Reader.kt

```diff
@@ -0,0 +1,17 @@
+package com.dropbox.external.store5
+
+import kotlinx.serialization.KSerializer
+
+/**
+ * Reads from [Market].
+ * @param fetcher Gets from remote data source.
+ * @see [Market].
+ * @see [Fetcher]
+ */
+data class Reader<Key : Any, Input : Any, Output : Any>(
```

Comment:
What is this abstraction for? I'm confused why `Fetcher` can't have the key/serializer/completion props.

Reply:
Reader and Writer interact with the Market. Fetcher and Updater interact with the remote data service